crates/integration_test/src/utils/test_utils.rs

```diff
@@ -64,62 +64,65 @@ pub fn create_container<P: AsRef<Path>>(id: &Uuid, dir: P) -> Result<Child> {
 }
 
 /// Sends a kill command to the given container process
-pub fn kill_container<P: AsRef<Path>>(id: &Uuid, dir: P) -> Result<Child> {
-    let res = Command::new(get_runtime_path())
-        .stdout(Stdio::piped())
-        .stderr(Stdio::piped())
-        .arg("--root")
-        .arg(dir.as_ref().join("runtime"))
+pub fn kill_container<P: AsRef<Path>>(id: &str, dir: P) -> Result<Child> {
+    let res = runtime_command(dir)
         .arg("kill")
-        .arg(id.to_string())
+        .arg(id)
         .arg("9")
         .spawn()
         .context("could not kill container")?;
     Ok(res)
 }
 
-pub fn delete_container<P: AsRef<Path>>(id: &Uuid, dir: P) -> Result<Child> {
-    let res = Command::new(get_runtime_path())
-        .stdout(Stdio::piped())
-        .stderr(Stdio::piped())
-        .arg("--root")
-        .arg(dir.as_ref().join("runtime"))
+pub fn delete_container<P: AsRef<Path>>(id: &str, dir: P) -> Result<Child> {
+    let res = runtime_command(dir)
         .arg("delete")
-        .arg(id.to_string())
+        .arg(id)
         .spawn()
         .context("could not delete container")?;
     Ok(res)
 }
 
-pub fn get_state<P: AsRef<Path>>(id: &Uuid, dir: P) -> Result<(String, String)> {
-    sleep(SLEEP_TIME);
-    let output = Command::new(get_runtime_path())
+pub fn get_state<P: AsRef<Path>>(id: &str, dir: P) -> Result<Child> {
+    let res = Command::new(get_runtime_path())
         .stdout(Stdio::piped())
         .stderr(Stdio::piped())
         .arg("--root")
         .arg(dir.as_ref().join("runtime"))
         .arg("state")
         .arg(id.to_string())
-        .spawn()?
-        .wait_with_output()?;
+        .spawn()
+        .context("could not get container state")?;
+    Ok(res)
+}
+
+pub fn get_state_output<P: AsRef<Path>>(id: &str, dir: P) -> Result<(String, String)> {
```

Comment:
Hey, I don't have any better ideas either, but can we change names of these functions? `get_state` name indicates that it will return state of the container, rather than the child process running `runtime state` command. I don't have any better suggestions as of now either, but ideally, the function name should indicate if its gonna run the command or get you the o/p

One suggestion is that, as far as I can see, `get_state` is only used in the state test, and if we can replace it with `get_state output` as per my comment on it, then we can make the current `get_state` private function of the module and not expose it, maybe name it `run_state_command` or something, and rename current `get_state_output` to `get_state`.